### PR TITLE
[API-733] Manual Activity missing from developers.strava.com

### PR DESCRIPTION
**Pull request by hariniriyer**
Added the Create Activity endpoint documentation for manual activities. Stubbed out all the identifying information (hopefully)

### CODE REVIEW

**Comment by hariniriyer** [comment]
image

**Comment by MateoAO** [comment]
Is that an actual response from a manual create? Surprising but makes sense

**Comment by hariniriyer** [comment]
@MateoAO Yes.

**Comment by MateoAO** [comment]
Any reason why the other input parameters were omitted? http://apidocs.strava.com/internal/v3/activities/#create

**Comment by hariniriyer** [comment]
So, good point. I meant to talk to you about that. If you look at the code changes, you will see that I added all the input parameters, but I think Swagger is stripping the optional ones on some basis. I tried using 'required: false' but that did not do the trick. I checked other endpoints which seem to show the same behavior.

**Comment by MateoAO** [comment]
Paging @zackisaacs

File: swagger/swagger.json.mustache [hunk 1]
```diff
@@ -1074,6 +1074,152 @@
...
+                "in": "path",
+                "description": "In seconds.",
+                "type": "integer",
+                "required": true
+            },
+            {
+                "name": "description",
+                "description": "Description of the activity.",
+                "type": "string",
+                "required": false
```
Comment (zackisaacs) [inline swagger/swagger.json.mustache]: The problem is that you need an `"in"` field for each parameter, e.g.
```
"in": "path"
```
It's also good practice to include the `required` field for all parameters, just to be explicit.

**Comment by hariniriyer** [comment]
image
@zackisaacs @MateoAO PTAL

File: swagger/swagger.json.mustache [hunk 2]
```diff
@@ -1074,6 +1074,163 @@
...
     },
     "/activities/{id}": {
+        "post": {
+            "operationId": "createActivity",
+            "summary": "Create an Activity",
+            "description": "Creates a manual activity for an athlete. Requires write permissions, as requested during the authorization process.",
+            "parameters": [
+            {
+               "name": "name",
+               "in": "path",
```
Comment (zackisaacs) [inline swagger/swagger.json.mustache]: I think we're going to want `"in": "body"` for all of these parameters. We'd use `"path"` for parameters like `activities/{id}` and `"query"` for parameters like `athlete?age=50`.

Here are the docs for [describing a request body parameter](https://swagger.io/docs/specification/2-0/describing-request-body/) using OpenAPI V2 (note that there is a V3... we want to get there eventually). The main thing to note is that you need to define a `"schema"` for request body parameters. I'd look at the first Object Payload example for specifics about how to define the schema.

Comment (hariniriyer) [reply swagger/swagger.json.mustache]: @zackisaacs Or we can have "in": "formData" ? There is some mismatch in the way the examples are laid out in swagger vs. what i see in the mustache file. Obviously the file types are different and I can look into it. But in the interest of time, I can make this a formData. The difference between the formData and body is the Content Type. In both cases, it is sent in the message body. Let me know what you think.

Comment (zackisaacs) [reply swagger/swagger.json.mustache]: Feel free to use `"formData"` for this request.

**Comment by hariniriyer** [comment]
@zackisaacs PTAL

File: swagger/swagger.json.mustache [hunk 3]
```diff
@@ -1073,7 +1073,166 @@
...
+            },
+            {
+                "name": "commute",
+                "in": "formData",
+                "description": "Set to 1 to mark as commute.",
+                "type": "integer",
+                "required": false
+            }
+
+
```
Comment (zackisaacs) [inline swagger/swagger.json.mustache]: Let's get rid of these two blank lines.

File: swagger/swagger.json.mustache [hunk 4]
```diff
@@ -1073,7 +1073,166 @@
...
+            },
+            "default": {
+                "description": "Unexpected error.",
+                "schema": {
+                    "$ref": "{{reference_prefix}}/fault.json#/Fault"
+                }
+            }
+        }
+        },
+
```
Comment (zackisaacs) [inline swagger/swagger.json.mustache]: Let's get rid of this blank line.

File: swagger/swagger.json.mustache [hunk 5]
```diff
@@ -1073,7 +1073,166 @@
...
+                }
+            },
+            "default": {
+                "description": "Unexpected error.",
+                "schema": {
+                    "$ref": "{{reference_prefix}}/fault.json#/Fault"
+                }
+            }
+        }
+        },
```
Comment (zackisaacs) [inline swagger/swagger.json.mustache]: Something's not quite right here -- there should never be two closing braces on the same line. I don't see the issue at first glance.

File: swagger/swagger.json.mustache [hunk 6]
```diff
@@ -1073,7 +1073,166 @@
         }
       }
     },
-    "/activities/{id}": {
+    "/activities": {
+        "post": {
```
Comment (zackisaacs) [inline swagger/swagger.json.mustache]: These indentations should all be 2 spaces instead of 4.

**Comment by hariniriyer** [comment]
@zackisaacs I am not using IntelliJ. However, I think I should. I will install it sometime today and fix the weirdness.

**Comment by hariniriyer** [comment]
@zackisaacs PTAL. Sorry I was too lazy to install IntelliJ. But thats the future.

**Pull request merged**